cp22/hpi-fun2/missing/executables/generator.py

```diff
@@ -0,0 +1,27 @@
+from asyncore import write
+import random
+from pathlib import Path
+import string
+
+random.seed(-5439687794124513171)
+
+
+def write_testcase(name, desc, string):
+    Path(name + '.in').write_text(string)
+    Path(name + '.desc').write_text(desc + '\n')
+
+max_length = 2500
+
+write_testcase("sample1", "A simple testcase", "allaoggo")
+write_testcase("sample2", "A simple testcase", "banana")
+write_testcase("sample3", "A simple testcase", "masseffect")
+some_text = "sqddbydscraqgvzfquizlmdopleiqtceyemhuyotzj"
+write_testcase("single_letter", "Only one letter", "x")
+write_testcase("long_palindrome", "A long palindrome", some_text + "".join(reversed(some_text)))
+write_testcase("no_palindromes", "A testcase without palindromes", "azbcdefgzhijklmznopqrsztuvw")
```

Comment:
Maybe add a dedicated "anti-greedy" case, which punishes you for taking the longest palindrome possible at every position like `abbacddcbbae`. Greedy takes abba, cddc, bb, a, e. Optimal is a, bbacddcbba, e.